api/openapi.yaml

```diff
@@ -2118,6 +2118,8 @@ components:
           type: string
         malwareType:
           $ref: '#/components/schemas/MalwareType'
+        ruleName:
+          type: string
```

Comment:
The name of this field is still bothering me, I don't like how specific it seems to be to yara and how much of a singleton it makes this object.

Remember if two malware scanners identify the same malware in the same file it should ideally be a single entry in the result not two separate ones otherwise all we're doing is smashing all the results together.

I think we need a structure like:

```
matches:
   type: array
   items:
      type: object
   properties:
      scanner:
         type: string
      signature:
         type: string
```

then each finding in the malware report is for a specific Malware and Path.

For example:

```
malwares: [
   {
      "malwareName": "trojan bad bad",
      "path": "/tmp/averybadfile",
      "matches": [
          {
             "scanner": "yara",
             "signature": "thetrojanbadbad_rule1"
          },
          {
              "scanner": "clamav",
              "signature": "unknown"
          },
      ]
   }
]
```

For a match which doesn't know the malwareName, then we can just use "unknown". For example:

```
   {
      "malwareName": "unknown",
      "path": "/tmp/averybadfile",
      "matches": [
          {
             "scanner": "yara",
             "signature": "somerule_thatdoesnt_list_themalware"
          },
      ]
   }
```

If there are multiple matches for the same path but we don't know the malware they can be merged together like:

```
   {
      "malwareName": "unknown",
      "path": "/tmp/averybadfile",
      "matches": [
          {
             "scanner": "yara",
             "signature": "somerule_thatdoesnt_list_themalware"
          },
          {
             "scanner": "yara",
             "signature": "another_rule_with_no_malware"
          },
      ]
   }
```

Reply:
I've opened an issue: https://github.com/openclarity/vmclarity/issues/677